index.js

```diff
@@ -1242,59 +1252,75 @@
-  //. [Just(1), Just(4), Nothing()]
+  //. See also [`get`](#get).
   //.
-  //. > S.pluck('x', [{x: 1}, {x: 2}, {x: undefined}])
-  //. [Just(1), Just(2), Just(undefined)]
+  //. ```javascript
+  //. > S.pluck(Number, 'x', [{x: 1}, {x: 2}, {x: '3'}, {x: null}, {}])
+  //. [Just(1), Just(2), Nothing(), Nothing(), Nothing()]
   //. ```
-  S.pluck = def('pluck', [String, Accessible], function(key, xs) {
-    return R.map(get(key), xs);
+  S.pluck = def('pluck', [Type, String, Accessible], function(type, key, xs) {
+    return R.map(get(type, key), xs);
   });
 
   //. ### Object
 
-  //# get :: String -> Object -> Maybe *
+  //# get :: Type -> String -> Accessible -> Maybe a
+  //.
+  //. Takes a type, a property name, and an object and returns Just the
+  //. value of the specified object property if it is of the specified
+  //. type (according to [`is`](#is)); Nothing otherwise.
+  //.
+  //. Since most values in the language inherit from Object, this type
+  //. acts a little like a wildcard.
   //.
-  //. Takes a property name and an object and returns Just the value of
-  //. the specified property of the object if the object has such an own
-  //. property; Nothing otherwise.
+  //. See also [`gets`](#gets).
   //.
   //. ```javascript
-  //. > S.get('x', {x: 1, y: 2})
+  //. > S.get(Number, 'x', {x: 1, y: 2})
   //. Just(1)
   //.
-  //. > S.get('toString', {x: 1, y: 2})
+  //. > S.get(Number, 'x', {x: '1', y: '2'})
+  //. Nothing()
+  //.
+  //. > S.get(Number, 'x', {})
   //. Nothing()
   //. ```
   var get = S.get =
-  def('get', [String, Accessible],
-      R.ifElse(R.has, R.compose(Just, R.prop), Nothing));
+  def('get', [Type, String, Accessible], function(type, key, obj) {
+    return filter(is(type), Just(obj[key]));
+  });
 
-  //# gets :: [String] -> Object -> Maybe *
+  //# gets :: Type -> [String] -> Accessible -> Maybe a
   //.
-  //. Takes a list of property names and an object and returns Just the
-  //. value at the path specified by the list of property names if such
-  //. a path exists; Nothing otherwise.
+  //. Takes a type, a list of property names, and an object and returns Just
+  //. the value at the path specified by the list of property names if such
+  //. a path exists and the value is of the specified type; Nothing otherwise.
+  //.
+  //. See also [`get`](#get).
   //.
   //. ```javascript
-  //. > S.gets(['a', 'b', 'c'], {a: {b: {c: 42}}})
+  //. > S.gets(Number, ['a', 'b', 'c'], {a: {b: {c: 42}}})
   //. Just(42)
   //.
-  //. > S.gets(['a', 'b', 'c'], {})
+  //. > S.gets(Number, ['a', 'b', 'c'], {a: {b: {c: '42'}}})
+  //. Nothing()
+  //.
+  //. > S.gets(Number, ['a', 'b', 'c'], {})
   //. Nothing()
   //. ```
-  S.gets = def('gets', [Accessible, Accessible], function(keys, obj) {
-    return R.reduce(function(acc, key) {
-      return R.chain(get(key), acc);
-    }, Just(obj), keys);
+  S.gets =
+  def('gets', [Type, Accessible, Accessible], function(type, keys, obj) {
```

Comment:
I'm a little behind on the whole Sanctuary type checking system but is there no List type?